src/interfaces/earnings.ts

```diff
@@ -156,7 +156,8 @@ export class EarningsInterface<C extends ChainId> extends ServiceInterface<C> {
       ? BigZero
       : assetsData
           .map((datum) => {
-            const apy = apys[datum.assetAddress]?.net_apy || 0;
+            const isNew = apys[datum.assetAddress]?.type === "new";
+            const apy = (!isNew && apys[datum.assetAddress]?.net_apy || 0)
```

Comment:
How about returning early if the apy is new? Something like?

```suggestion
            if (apys[datum.assetAddress]?.type === "new") {
              return BigZero;
            }
            const apy = apys[datum.assetAddress]?.net_apy || 0;
```

It would be good to have a test for this, I was actually working on a PR for this and wrote one test. You can try and write one, or just replace the whole `describe("when there is an account", () => {}` with the following:

```typescript
    describe("when there is an account", () => {
      const accountEarningsResponse = createMockAccountEarningsResponse();
      const tokensValueInUsdcMock: jest.Mock<Promise<BigNumber>> = jest.fn();

      beforeEach(() => {
        subgraphFetchQueryMock.mockResolvedValue(accountEarningsResponse);
        (earningsInterface as any).tokensValueInUsdc = tokensValueInUsdcMock;
        tokensValueInUsdcMock.mockResolvedValue(new BigNumber(10));
        getAddressMock.mockReturnValue("0x001");
      });

      it("should return an empty EarningsUserData", async () => {
        const accountAddress: Address = "0x001";
        visionApyMock.mockResolvedValue({
          [accountAddress]: createMockApy({
            net_apy: 42,
          }),
        });
        getAddressMock.mockReturnValue(accountAddress);

        const actual = await earningsInterface.accountAssetsData(accountAddress);

        expect(visionApyMock).toHaveBeenCalledTimes(1);
        expect(visionApyMock).toHaveBeenCalledWith([accountAddress]);
        expect(getAddressMock).toHaveBeenCalledTimes(1);
        expect(actual).toEqual({
          earnings: "10",
          earningsAssetData: [
            {
              assetAddress: accountAddress,
              earned: "10",
            },
          ],
          estimatedYearlyYield: "420",
          grossApy: 42,
          holdings: "10",
        });
      });

      it("should filter out new vaults for the estimated yield calculation", async () => {
        const accountAddress: Address = "0x001";
        visionApyMock.mockResolvedValue({
          [accountAddress]: createMockApy({
            net_apy: 42,
            type: "new",
          }),
        });

        const actual = await earningsInterface.accountAssetsData(accountAddress);

        expect(visionApyMock).toHaveBeenCalledTimes(1);
        expect(visionApyMock).toHaveBeenCalledWith([accountAddress]);
        expect(getAddressMock).toHaveBeenCalledTimes(1);
        expect(actual).toMatchObject({
          estimatedYearlyYield: "0",
        });
      });
    });
``

Reply:
1st part addressed with https://github.com/yearn/yearn-sdk/pull/281/commits/47405454ad70ace3c3882064f26b6c26855a045c

Tests added with https://github.com/yearn/yearn-sdk/pull/281/commits/6efcadae3c9370e63b0880065165858f0d7e2cff